### DISCUSSION TITLE
Can I add a video to a scene?

### DISCUSSION THREAD

**Discussion by F3RN4NDOJF3RN4ND3Z** [StereoKit/StereoKit#250] [Q&A]
I want to add a video in a UI windows I see that there is only image. Is it possible?

**Comment by maluoi** [comment]
StereoKit doesn't currently have built-in video support! It's on the roadmap, but in the meantime that's something you'll have to implement on your side. In theory, you _can_ decode a video with some external library, and upload the video's pixel data into a texture. I haven't tried this before, so it could be some effort, but it should work!

**Reply by F3RN4NDOJF3RN4ND3Z** [reply]
If you have some example code I can follow it could be great, I think you mean with decode like ffmpeg? but with it whe only have image so I can stream those images to a texture in each frame and and see like a video looping but with no sound

**Reply by maluoi** [reply]
Yeah, something like ffmpeg. I haven't tried this before, so I unfortunately don't have any sample code on-hand!

For audio, StereoKit does also support audio-streams, see [`Sound.CreateStream`](https://stereokit.net/Pages/Reference/Sound/CreateStream.html), and you may be able to use that as well to play the video's audio.

I do think this is all likely to be a decent chunk of work, so I don't recommend tackling it unless you have the skills and the time. It could be more effective to find an alternative route to achieving your goal, at least until I eventually find the time to add this functionality into SK's core.

**Reply by TopperDEL** [reply]
This is something I would Nee, too, next year. So a generic way of doing that would be great. I would also contribute to this if possible.

How about a StereoKit.Media-package that adds the dependencies like ffmpeg to the project so the core does not size up?
Using ffmpeg should be quite easy. My main concern would be the translation into a texture and the correct mapping to the frames per second.

**Reply by maluoi** [reply]
My current thought is more along the lines of DirectShow and/or Vulkan Video. That would help ensure hardware acceleration and texture compatibility, and would probably be fine even in StereoKit's core :)

Ffmpeg is probably more of an "on top" solution, but could also work, possibly without even much modification to StereoKit? I'd need to dig a bit deeper here to know for certain.

**Reply by cdaniel09** [reply]
vulkan video is going to require ffmpeg as part of the integration i believe, did some research tried to finish a video project but not going to happen. Hopefully I'll pick it up again and get something running

**Reply by maluoi** [reply]
Ah, I didn't realize Vulkan Video wasn't a standalone thing! But I guess it makes some sense. Yeah, that could be a little problematic 😬 Thanks for highlighting it though, that's super helpful to know.
